src/GraphQL.Dynamic/LiteralGraphType/RemoteLiteralGraphType.cs

```diff
@@ -0,0 +1,344 @@
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Reflection;
+using System.Reflection.Emit;
+using System.Threading.Tasks;
+using GraphQL.Dynamic.Types.Introspection;
+using GraphQL.Introspection;
+using GraphQL.Types;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace GraphQL.Dynamic.Types.LiteralGraphType
+{
+    public class RemoteLiteralGraphType : ObjectGraphType
+    {
+        public delegate Introspection.Schema RemoteSchemaFetcher(string url);
+
+        private static object FetchRemoteServerSchemaMutex = new object();
+        private static ConcurrentDictionary<string, Introspection.Schema> RemoteServerSchemas { get; set; } = new ConcurrentDictionary<string, Introspection.Schema>();
+
+        private static ConcurrentDictionary<string, HashSet<Type>> RemoteServerTypes { get; set; } = new ConcurrentDictionary<string, HashSet<Type>>();
+
+        private bool _hasAddedFields;
+
+        private readonly string _remoteLocation;
+        private readonly string _name;
+
+        public RemoteLiteralGraphType(string remoteLocation, string name)
+        {
+            _remoteLocation = remoteLocation;
+            _name = name;
+        }
+
+        public override string CollectTypes(TypeCollectionContext context)
+        {
+            if (!string.IsNullOrWhiteSpace(_remoteLocation) && !string.IsNullOrWhiteSpace(_name))
+            {
+                // get remote server information
+                // if we haven't fetched the remote types for this remote:
+                if (!RemoteServerSchemas.TryGetValue(_remoteLocation, out var schema))
+                {
+                    throw new Exception($"Schema not already loaded for remote {_remoteLocation}");
+                }
+
+                // get type
+                if (!TryGetFieldTypeFromRemoteSchema(schema, _name, out var type))
+                {
+                    // if no type found: fail
+                    // TODO: fail better
+                    throw new Exception($"Failed to find type '{_name}' in remote '{_remoteLocation}' schema");
+                }
+
+                Name = $"{_remoteLocation}.{_name}";
```

Comment:
this generates not valid names. changing to ```Name = $"{_name}";``` should fix this. or is there a reason other than duplicated types for this?

Reply:
yep; you've got it -- it's to avoid name collisions.  This should be fixed as part of merging my changes in!